### PR TITLE
test(checkbox): playwright refactor

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Checkbox tests from Cypress to Playwright

### Current behaviour

Tests are currently Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [x] Run Checkbox tests in Playwright Test Runner to check if the `checkbox.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/checkbox/checkbox.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,1052 @@
...
+    }) => {
+      await mount(<CheckboxComponent data-component={CHARACTERS.STANDARD} />);
+
+      const checkboxElement = checkboxRole(page);
+      await checkboxElement.focus();
+      const focusedElement = checkboxRole(page).locator("..").locator("div");
+      await expect(focusedElement).toHaveCSS(
+        "box-shadow",
+        "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+      );
```
Comment (sianford) [inline src/components/checkbox/checkbox.pw.tsx:62]: Do we not need to test the outline here too as in the Cypress test?

File: src/components/checkbox/checkbox.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,1052 @@
...
+      <CheckboxComponent
+        label="Label For CheckBox"
+        labelHelp="Label Help"
+        helpAriaLabel="This text provides more information for the label"
+      />
+    );
+
+    const checkboxIconElement = checkboxIcon(page);
+    await checkboxIconElement.hover();
+    const checkbooxHelpElement = await checkboxHelpIcon(page);
```
Comment (sianford) [inline src/components/checkbox/checkbox.pw.tsx:172]: const should read `checkboxHelpElement` but it reads `checkbooxHelpElement`.

File: src/components/checkbox/checkbox.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,1052 @@
...
+    await expect(checkbooxHelpElement).toHaveAttribute(
```
Comment (sianford) [inline src/components/checkbox/checkbox.pw.tsx:173]: Amend to `checkboxHelpElement` from `checkbooxHelpElement` once the const has been amended.

**Review by sianford — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**